src/validation.ts

```diff
@@ -1,43 +1,35 @@
-import { 
-  PayloadType,
-  ContractCallPayload,
-  getAbi,
-  addressToString,
-  StacksTransaction
-} from "@stacks/transactions";
+import { PayloadType, ContractCallPayload, getAbi, addressToString, StacksTransaction } from '@stacks/transactions';
 
 // rules for transaction sponsorship eligibility
 // customize this if you are forking this repo
 export async function validateTransaction(transaction: StacksTransaction) {
   if (transaction.payload.payloadType !== PayloadType.ContractCall) {
     throw new Error('Transaction is not a contract call');
-  } 
+  }
 
   const payload = transaction.payload as ContractCallPayload;
 
   // check and limit to SIP-09 transfers calls
-  if (payload.functionName.content.toString() !== 'transfer') {
-    throw new Error('Transaction is not a NFT transfer contract call');
-  } 
+  if (payload.contractAddress.hash160.toString() !== 'e685b016b3b6cd9ebf35f38e5ae29392e2acd51d') {
```

Comment:
Why do we need to hash the contract address instead of just comparing directly?

Reply:
@yknl I was inspecting the signed transaction payload, and this is what was already being broadcasted: 
<img width="655" alt="image" src="https://github.com/secretkeylabs/stacks-transaction-sponsor/assets/6109710/0645073b-7ed0-4797-8799-9d88b2f5bc8b">

do you mean, it would be better to compare the addressToString output on L17?

Reply:
Yes, much more straight forward to compare the address string. Also, the hash is not the complete address. It doesn't include the version.